aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2208,6 +2009,82 @@ Tensor cholesky_inverse(const Tensor &input, bool upper) {
   return result;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ linalg.solve ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+// Auxiliary function that returns the LU decomposition to use it in the backward
+TORCH_IMPL_FUNC(_linalg_solve_out)(const Tensor& A,
+                                   const Tensor& B,
+                                   bool left,
+                                   const Tensor& result,
+                                   const Tensor& LU,
+                                   const Tensor& pivots) {
+  // Possible optimization: Compute the LU factorization of A^T if A is contiguous
+  // Then we solve A^T X = B with adjoint=True
+  // This saves a copy as A doesn't need to be copied into an F-contig matrix in lu_factor
+  const bool use_A_T = A.is_contiguous() && !A.is_complex();
+  auto info = at::empty({0}, A.options().dtype(kInt));
+  at::linalg_lu_factor_ex_out(const_cast<Tensor&>(LU),
+                              const_cast<Tensor&>(pivots),
+                              const_cast<Tensor&>(info),
+                              use_A_T ? A.mT() : A,
+                              /*pivot=*/true,
+                              /*check_errors=*/false);
+  at::_linalg_check_errors(info, "torch.linalg.solve", A.dim() == 2);
```

Comment:
Shall we have a version without error checks? CC @IvanYashchuk , @mruberry .

Reply:
It could be of interest. The funciton is equivalent ot `lu_factor_ex` + `lu_solve`, but the derivative of `linalg_solve` is simpler than those of those two combined.